repos/system_upgrade/el7toel8/actors/redhatsignedrpmscanner/actor.py

```diff
@@ -26,16 +29,43 @@ def process(self):
         signed_pkgs = InstalledRedHatSignedRPM()
         unsigned_pkgs = InstalledUnsignedRPM()
 
+        env_vars = self.configuration.leapp_env_vars
+        # if we start upgrade with LEAPP_DEVEL_RPMS_ALL_SIGNED=1, we consider
+        # all packages to be signed
+        all_signed = [
+            env
+            for env in env_vars
+            if env.name == 'LEAPP_DEVEL_RPMS_ALL_SIGNED' and env.value == '1'
+        ]
+
+        def some_rhsigs_keys_in_pkg_pgpsig(pkg):
+            return any(key in pkg.pgpsig for key in RH_SIGS)
+
+        def has_gpg_pubkey(pkg):
+            """Check if gpg-pubkey pkg exists or LEAPP_DEVEL_RPMS_ALL_SIGNED=1
+
+            gpg-pubkey is not signed as it would require another package
+            to verify its signature
+            """
+            return (    # pylint: disable-msg=consider-using-ternary
+                    pkg.name == 'gpg-pubkey'
+                    and pkg.packager.startswith('Red Hat, Inc.')
+                    or all_signed
+            )
+
+        def has_katello(pkg):
+            """Whitelist the katello package."""
+            return pkg.name.startswith('katello')
```

Comment:
@ZhukovGreen @oamg/developers  We should test as long prefi as possible, in this case, as far as I understand it should be `katello-ca-consumer`, not `katello`, Or do we know any others katello rpms that we want to whitelist?

Reply:
@pirat89 I don't know any other katello package. I will extend this prefix match

Reply:
Fixed, thank you @pirat89